getnacelle/nacelle-js
# feat: add `productCollectionEntries` method

## Why are these changes introduced?

Addresses [ENG-8326](https://nacelle.atlassian.net/browse/ENG-8326) 

Adds the `productCollectionEntries` method



## What is this pull request doing?

- Adds the `productCollectionEntries` method to `@nacelle/storefront-sdk-plugin-commerce-queries` to match `@nacelle/storefront-sdk@1.x`'s productCollectionEntries method.



## How to Test



- Check out the `ENG-8178/add-productCollectionEntries-method` branch.
- Navigate to `packages/storefront-sdk-plugins/commerce-queries`.
- npm run coverage - all tests should pass with 100% coverage.
<img width="661" alt="Screen Shot 2023-02-02 at 9 40 21 AM" src="https://user-images.githubusercontent.com/25146108/216370549-66290d13-b7cb-4d4a-b9c9-50a41360ddb2.png">
- `npm run build` - there should be no build errors.

#### Local
- Test locally with `npm link`:
```js
import {StorefrontClient} from '@nacelle/storefront-sdk'
import commerceQueriesPlugin from '@nacelle/storefront-sdk-plugin-commerce-queries'

const storefrontEndpoint = "https://storefront.api.development.nacelle.com/graphql/v1/spaces/<SPACE_ID>"
const ClientWithCommerceQueries = commerceQueriesPlugin(StorefrontClient);
const client = new ClientWithCommerceQueries({ storefrontEndpoint });


const entries = await client.productCollectionEntries({handle: '<HANDLE>'})
```
- Make sure the code works as expected
<img width="208" alt="Screen Shot 2023-02-02 at 9 43 23 AM" src="https://user-images.githubusercontent.com/25146108/216371400-0953f85c-05f0-4cc6-8ac7-bf958419b2a2.png">




## Checklist



[ENG-8326]: https://nacelle.atlassian.net/browse/ENG-8326?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
